src/GovUk.Education.ExploreEducationStatistics.Common.Tests/Cache/CacheAspectThreadManagementIntegrationTests.cs

```diff
@@ -0,0 +1,589 @@
+
+            var resultToCache = new ControllerResponse(key);
+            _eventList.AddEvent($"Controller method with key {key} completed after delay - returning {resultToCache}");
+            return resultToCache;
+        }
+    }
+
+    /// <summary>
+    /// This TestBlobCacheService adds an artificial delay to its GetItemAsync and SetItemAsync methods to simulate some
+    /// async work e.g. reading and writing from Azure Blob Storage.
+    /// </summary>
+    private class TestBlobCacheService : IBlobCacheService
+    {
+        public readonly ConcurrentDictionary<TestCacheKey, ControllerResponse> Cache = new();
+        private readonly EventList _eventList;
+
+        public TestBlobCacheService(EventList eventList)
+        {
+            _eventList = eventList;
+        }
+        
+        public object GetItem(IBlobCacheKey cacheKey, Type targetType)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task<object?> GetItemAsync(IBlobCacheKey cacheKey, Type targetType)
+        {
+            _eventList.AddEvent($"Requesting cached item with key {cacheKey}");
+            
+            var key = cacheKey as TestCacheKey;
+            await Task.Delay(AsyncOperationDelayMillis);
+            
+            if (Cache.ContainsKey(key!))
+            {
+                var result = Cache[key!];
+                _eventList.AddEvent($"Returning cached item {result} with key {cacheKey} after delay");
+                return result;
+            }
+            
+            _eventList.AddEvent($"Returning no cached item with key {cacheKey} after delay");
+            return null;
+        }
+
+        public void SetItem<TItem>(IBlobCacheKey cacheKey, TItem item)
+        {
+            throw new NotImplementedException();
+        }
+
+        public async Task SetItemAsync<TItem>(IBlobCacheKey cacheKey, TItem item)
+        {
+            _eventList.AddEvent($"Setting cached item {item} with key {cacheKey}");
+            
+            var result = item as ControllerResponse;
+            Cache[(cacheKey as TestCacheKey)!] = result!;
+            await Task.Delay(AsyncOperationDelayMillis);
+            
+            _eventList.AddEvent($"Set cached item {item} with key {cacheKey} after delay");
+        }
+
+        public Task DeleteItemAsync(IBlobCacheKey cacheKey)
+        {
+            throw new NotImplementedException();
+        }
+
+        public Task DeleteCacheFolderAsync(IBlobCacheKey cacheFolderKey)
+        {
+            throw new NotImplementedException();
+        }
+    }
+
+    private class EventList
+    {
+        private readonly List<Event> _events = new();
+        
+        public void AddEvent(string message)
+        {
+            _events.Add(new Event(message));
+        }
+
+        public List<Event> GetEvents()
+        {
+            return _events;
+        }
+
+        public void Clear()
+        {
+            _events.Clear();
+        }
+    }
```

Comment:
Any need to restrict the methods of `EventList` from that of a regular `List`? Just wondering if this would work?

```
    public class EventList : List<Event>
    {
    }
```

Reply:
Hey Ben - good call.  This as with a few other things in this PR had evolved from something simple into something more complex before then going back to something simple again!

I've replaced the whole lot with even simpler `var events = new List<Event>()` all over.